hack/boilerplate/boilerplate.go

```diff
@@ -0,0 +1,152 @@
+	rootdir        *string
+	boilerplatedir *string
+)
+
+func init() {
+	cwd, _ := os.Getwd()
+	boilerplatedir = flag.String("boilerplate-dir", cwd, "Boilerplate directory for boilerplate files")
+	cwd = cwd + "/../../"
+	rootdir = flag.String("rootdir", filepath.Dir(cwd), "Root directory to examine")
+}
+
+func main() {
+	flag.Parse()
+	refs := getRefs(*boilerplatedir)
+	if len(refs) == 0 {
+		log.Fatal("no references in ", *boilerplatedir)
+	}
+	files := getFileList(*rootdir, refs)
+	for _, file := range files {
+		if !filePasses(file, refs[getFileExtension(file)]) {
+			fmt.Println(file)
+		}
+	}
+
+}
+
+/*
+This function is to populate the refs variable with the
+different boilerplate/template for different extension.
+*/
+func getRefs(dir string) map[string][]byte {
+	refs := make(map[string][]byte)
+	files, _ := filepath.Glob(dir + "/*.txt")
+	for _, filename := range files {
+		extension := strings.ToLower(strings.Split(filename, ".")[1])
+		data, err := ioutil.ReadFile(filename)
+		if err != nil {
+			log.Fatal(err)
+		}
+		re := regexp.MustCompile(`\r`)
+		refs[extension] = re.ReplaceAll(data, nil)
+	}
+	return refs
+}
+
+/*
+Function to check whether the processed file
+is valid.
+Returning false means that the file does not the
+proper boilerplate template
+*/
+func filePasses(filename string, ref []byte) bool {
+	var re *regexp.Regexp
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+	re = regexp.MustCompile(`\r`)
+	data = re.ReplaceAll(data, nil)
+
+	extension := getFileExtension(filename)
+
+	// remove build tags from the top of Go files
+	if extension == "go" {
+		re = regexp.MustCompile(`(?m)^(// \+build.*\n)+\n`)
+		data = re.ReplaceAll(data, nil)
+	}
+
+	// remove shebang from the top of shell files
+	if extension == "sh" {
+		re = regexp.MustCompile(`(?m)^(#!.*\n)\n*`)
+		data = re.ReplaceAll(data, nil)
+	}
+
+	// if our test file is smaller than the reference it surely fails!
+	if len(data) < len(ref) {
+		return false
+	}
+
+	data = data[:len(ref)]
+
+	// Search for "Copyright YEAR" which exists in the boilerplate, but shouldn't in the real thing
+	re = regexp.MustCompile(`Copyright YEAR`)
+	if re.Match(data) {
+		return false
+	}
+
+	// Replace all occurrences of the regex "Copyright \d{4}" with "Copyright YEAR"
+	re = regexp.MustCompile(`Copyright \d{4}`)
+	data = re.ReplaceAll(data, []byte(`Copyright YEAR`))
+
+	return bytes.Equal(data, ref)
+}
+
+/**
+Function to get the file extensin or the filename if the file has no extension
+*/
+func getFileExtension(filename string) string {
+	splitted := strings.Split(filepath.Base(filename), ".")
+	return strings.ToLower(splitted[len(splitted)-1])
+}
+
+/**
+Function to get all the files from the directory that heeds to be checked.
+*/
+func getFileList(rootDir string, extensions map[string][]byte) []string {
```

Comment:
in golang we dont use get in naming,
consider a better name, like "filesToCheck(..."

Reply:
Thanks for the review. I renamed the function.